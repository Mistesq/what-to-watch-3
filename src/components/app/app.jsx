import React, {PureComponent} from 'react';
import PropTypes from "prop-types";
import Main from '../main/main.jsx';

const movieTitleClickHandler = () => {};

class App extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Main
        promoFilm={this.props.promoFilm}
        movieList={this.props.movieList}
        onMovieTitleClick={movieTitleClickHandler}
      />
    );
  }
}

App.propTypes = {
  promoFilm: PropTypes.exact({
    filmName: PropTypes.string.isRequired,
    filmGenre: PropTypes.string.isRequired,
    filmYear: PropTypes.number.isRequired
  }).isRequired,
  movieList: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired,
    previewImage: PropTypes.string.isRequired,
    previewSrc: PropTypes.string.isRequired
  }))
};

export default App;
