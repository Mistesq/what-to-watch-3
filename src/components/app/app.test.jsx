import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {App} from "./app.jsx"

const mockStore = configureStore([]);

const ALL_GENRES = `All genres`;

const movieTestList = [
  {
    id: 0,
    title: `The Grand Budapest Hotel`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 240,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 20,
    comments: [
      {
        id: 0,
        text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        author: `Kate Muir`,
        date: `December 24, 2016`,
        rating: `8,9`
      },
      {
        id: 1,
        text: `Andersons films are too precious for some, but for those of us willing to
        lose ourselves in them, theyre a delight. The Grand Budapest Hotel is no different, except that he
        has added a hint of gravitas to the mix, improving the recipe.`,
        author: `Bill Goodykoontz`,
        date: `November 18, 2015`,
        rating: `8,0`
      }
    ],
  },
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: 2011,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8,
    ratingCount: 1240,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.`,
    director: `director2`,
    starring: [`actor`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 30,
    comments: [
      {
        id: 0,
        text: `I didnt find it amusing, and while I can appreciate the creativity, its an
        hour and 40 minutes I wish I could take back.`,
        author: `Amanda Greever`,
        date: `November 18, 2015`,
        rating: `8,0`
      },
    ],
  }
];

const promoFilmTest = {
  filmName: `The Wizard of Oz`,
  filmGenre: `Adventure`,
  filmYear: 1939
};

it(`Render App`, () => {
  const tree = renderer
  .create(<Provider store={store}>(<App promoFilm={promoFilmTest} films={movieTestList} />
    </Provider>, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
