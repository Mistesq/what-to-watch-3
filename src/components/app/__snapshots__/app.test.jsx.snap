// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
<Main
  films={
    Array [
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "comments": Array [
          Object {
            "author": "Kate Muir",
            "date": "December 24, 2016",
            "id": 0,
            "rating": "8,9",
            "text": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
          },
          Object {
            "author": "Bill Goodykoontz",
            "date": "November 18, 2015",
            "id": 1,
            "rating": "8,0",
            "text": "Andersons films are too precious for some, but for those of us willing to
        lose ourselves in them, theyre a delight. The Grand Budapest Hotel is no different, except that he
        has added a hint of gravitas to the mix, improving the recipe.",
          },
        ],
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
        "director": "Wes Andreson",
        "genre": "Drama",
        "id": 0,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 240,
        "ratingScore": 8.9,
        "releaseDate": 2014,
        "runTime": 20,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
        ],
        "title": "The Grand Budapest Hotel",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "comments": Array [
          Object {
            "author": "Amanda Greever",
            "date": "November 18, 2015",
            "id": 0,
            "rating": "8,0",
            "text": "I didnt find it amusing, and while I can appreciate the creativity, its an
        hour and 40 minutes I wish I could take back.",
          },
        ],
        "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "director": "director2",
        "genre": "Comedy",
        "id": 1,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 1240,
        "ratingScore": 8,
        "releaseDate": 2011,
        "runTime": 30,
        "starring": Array [
          "actor",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
        ],
        "title": "Bohemian Rhapsody",
      },
    ]
  }
  onCardClick={[Function]}
  promoFilm={
    Object {
      "filmGenre": "Adventure",
      "filmName": "The Wizard of Oz",
      "filmYear": 1939,
    }
  }
/>
`;
